### ISSUE TITLE
[Ch 21·23][EPUB][Major] 인용구 안 표가 tex·EPUB 에서 문단으로 뭉개짐 — 표 앞 빈 줄이 없으면 pandoc gfm 이 표로 안 읽음

### ISSUE THREAD

**Issue by fluentmin** [yoon-gu/neuqes-101#217]
> **요약** — 노트북에서 표 바로 앞줄에 문단이 붙어 있으면(사이에 빈 줄이 없으면), `notebook_to_tex.py` 가 쓰는 pandoc `gfm` 리더가 그걸 표로 읽지 않고 `\textbar{}` 나열 문단으로 뭉갭니다. Colab 은 관대하게 표로 그려 주기 때문에 노트북 화면에서는 멀쩡해 보이고, **tex·EPUB 에서만** 깨집니다. **Ch 21·Ch 23** 의 "숫자로 감 잡기" 표 2개가 여기 해당합니다.

## 무엇이 문제인가

원천 노트북(`23_ko_bert_classify.ipynb` cell 3)은 인용구 안에 표를 둡니다. 앞줄과 표 사이에 빈 줄이 없습니다.

```markdown
> **숫자로 감 잡기** (K=2, 정답 = 클래스 1):
> | logits $(z_0, z_1)$ | softmax → $\hat p_1$ | 손실 |
> |---|---|---|
> | (0, 0) | 0.5 | **0.693** ← random |
```

<img width="1568" height="768" alt="Image" src="https://github.com/user-attachments/assets/cdbe1419-5914-40ff-b5fd-a02f064fd888" />

생성된 tex(`book/chapters/ch23_ko_bert_classify.tex:262-268`, master `4b309eb`)에서는 표가 사라집니다.

```latex
\begin{quote}
\textbf{숫자로 감 잡기} (K=2, 정답 = 클래스 1):
\textbar{} logits \((z_0, z_1)\) \textbar{} softmax \(\to\) \(\hat p_1\) \textbar{} 손실 \textbar{}
\textbar-\/-\/-\textbar-\/-\/-\textbar-\/-\/-\textbar{}
\textbar{} (0, 0) \textbar{} 0.5 \textbar{} \textbf{0.693} \(\leftarrow\) random \textbar{}
```

EPUB 빌드는 이 tex 를 충실히 옮기므로 독자는 4행 3열 표 대신 `--------- (0, 0) 0.5 0.693 ← random (-1, +1) 0.881 0.127 ...` 이 한 문단으로 흘러가는 것을 읽게 됩니다. 구분행 `|---|---|` 까지 본문에 그대로 노출됩니다.

원인은 인용구가 아니라 **표 앞 빈 줄**입니다. 같은 리더(`gfm+tex_math_dollars+pipe_tables`, pandoc 2.19.2)로 네 갈래를 최소 재현했습니다.

| 케이스 | 결과 |
|---|---|
| 인용구 + 표 앞 빈 줄 **없음** (= 실제 노트북) | ❌ `\textbar-\/-\/-\textbar` 로 뭉개짐 |
| 인용구 + 표 앞 빈 줄 **있음** (`>` 만 있는 줄) | ✅ `longtable` 정상 |
| 인용구 아님 + 표 앞 빈 줄 없음 | ❌ 똑같이 뭉개짐 |
| 헤딩(`###`) 바로 다음 표, 빈 줄 없음 | ✅ 정상 (헤딩이 블록을 끊음) |

재현본의 `\textbar-\/-\/-\textbar` 는 실제 tex 264행과 글자 단위로 일치합니다.

**영향 범위** — 노트북 34개를 전수 스캔(표 헤더행 + 다음 줄이 구분행 + 앞줄이 비어 있지 않음)한 결과 **2개 장**입니다.

| 장 | 노트북 | 셀 | tex |
|---|---|---|---|
| **Ch 21** | `21_en_bert_classify.ipynb` | cell 3 | `ch21_en_bert_classify.tex:246-249` |
| **Ch 23** | `23_ko_bert_classify.ipynb` | cell 3 | `ch23_ko_bert_classify.tex:262-268` |

생성된 tex 전체에서 `\textbar-\/-\/-\textbar` 시그니처를 가진 파일도 이 둘뿐입니다. 스캔에 함께 걸린 `29_benchmark_eval` cell 29 와 `appendix_eval_landscape` cell 13 은 앞줄이 `###` 헤딩이라 정상 변환됩니다(오탐).

## 왜 고쳐야 하나

이 표는 두 장 모두 **📐 Loss 함수의 변화** 절의 핵심 교보재입니다. logits → softmax → 손실이 어떻게 대응하는지 4행을 나란히 놓고 읽어야 `0.693` 이 random baseline 이고 `4.018` 이 "자신 있게 틀림" 이라는 대비가 보입니다. 한 문단으로 뭉개지면 그 대응이 사라져 숫자 나열만 남습니다.

더 위험한 건 **조용히 깨진다**는 점입니다. Colab 에서는 정상으로 보이므로 노트북 검수만으로는 절대 발견되지 않고, 앞으로 같은 형태로 표를 쓰면 또 같은 일이 벌어집니다.

## 제안

노트북 2곳에 빈 줄을 넣는 것으로도 당장은 해결되지만, 재발을 막으려면 변환기 층에서 처리하는 편이 낫다고 봅니다. `escape_table_math_pipes()` 같은 표 전처리가 이미 있어 자리도 자연스럽습니다. 다만 표를 오인식하지 않도록 조건을 좁게 잡아야 합니다 — 다음 줄이 구분행일 때만 표의 시작으로 인정하고, 코드펜스 안은 건드리지 않고, 인용구 안에서는 접두사를 살린 빈 줄을 넣어 표가 인용구 밖으로 떨어져 나가지 않게 하는 정도입니다.

표가 하나 늘면 그 장의 캡션이 새로 겹치는 부작용이 있습니다 — 별도 이슈로 분리했습니다.

## 재현

```bash
git checkout 4b309eb
grep -n -A6 '숫자로 감 잡기' book/chapters/ch23_ko_bert_classify.tex
grep -rlF '\textbar-\/-\/-\textbar' book/chapters/

# 최소 재현 (pandoc 2.19.2)
printf '> **숫자로 감 잡기** (K=2):\n> | a | b |\n> |---|---|\n> | 1 | 2 |\n' \
  | pandoc -f gfm+tex_math_dollars+pipe_tables -t latex
```

기준 커밋 `4b309eb` (master) / pandoc 2.19.2 (tex 생성), 3.9.0.2 (EPUB 빌드), macOS

전파: Notebook → tex → EPUB (WikiDocs 는 마크다운을 직접 쓰므로 해당 없음)

## Related

Related to #64. 캡션 중복 부작용은 #218 로 분리했습니다.

수정 작업 중인 브랜치: `fix/tex-table-blank-line` (`75b7395`) — `insert_table_blank_lines()` 추가 + Ch 21·23 tex 재생성.

<img width="1568" height="768" alt="Image" src="https://github.com/user-attachments/assets/76c80359-b563-49b1-968f-242d6f3b2aab" />
